.github/workflows/compilation.yml

```
@@ -12,73 +16,70 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [macos-latest, ubuntu-latest, ubuntu-18.04]
+        os: [ ubuntu-latest, macos-latest ]
 
     steps:
-    - uses: actions/checkout@v2
-    
-    - name: Install Ubuntu dependencies
-      if: matrix.os == 'ubuntu-latest'
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Setup environment variables
       run: |
-        sudo apt-get update
-        sudo apt-get -y install texinfo bison flex findutils ninja-build
-        
-    - name: Install Ubuntu 18.04 dependencies
-      if: matrix.os == 'ubuntu-18.04'
+        # export first to set globally for $path
+        export ORBISDEV=$PWD/orbisdev
+        echo "ORBISDEV=$PWD/orbisdev"   >> $GITHUB_ENV
+        echo "PATH=$ORBISDEV/bin:$PATH" >> $GITHUB_ENV
+
+    - name: Install Ubuntu dependencies
+      if: runner.os == 'Linux'
       run: |
-        sudo apt-get update
-        sudo apt-get -y install texinfo bison flex findutils ninja-build
-    
+        wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+        sudo dpkg -i packages-microsoft-prod.deb
+        sudo apt update
+        sudo apt install libgmp-dev bison flex make texinfo wget m4 cmake findutils ninja-build clang g++ dotnet-runtime-3.1 dotnet-sdk-3.1
+
     - name: Install Mac dependencies
-      if: matrix.os == 'macOS-latest'
+      if: runner.os == 'macOS'
       run: |
         brew update
         brew install texinfo bison flex ninja
 
-    - name: Runs all the stages in the shell
-      run: |
-        export ORBISDEV=$PWD/orbisdev
-        export PATH=$ORBISDEV/bin:$PATH
-        ./build-all.sh
+    - name: Build stage 1
```

why are you running here in separate stages?

---

The time to run the individual stages can be monitored and if anything fails, it's easier to fix them instead of going through thousands line of log if running them all at once.

---

Nah.... when it fails it will show at the very end of the log what stage failed anyway.

---

Ok I'll revert it